File: generate-grades.js
const fs = require('fs');

// Leer estudiantes del archivo CSV
const usersFile = fs.readFileSync('users-consolidated-2025-CORREGIDO.csv', 'utf8');
const usersLines = usersFile.split('\n').slice(1).filter(line => line.trim());

const students = usersLines.map(line => {
  const parts = line.split(',');
  return {
    nombre: parts[1],
    rut: parts[2],
    curso: parts[6],
    seccion: parts[7]
  };
});

console.log(`📚 Cargados ${students.length} estudiantes del archivo`);

// Actividades para 1° Básico
const actividades1ro = [
  { asignatura: "Matemáticas", profesor: "Prof. Ana Torres", fecha: "2025-03-15", tipo: "prueba", tema: "Números hasta el 20" },
  { asignatura: "Lenguaje", profesor: "Prof. Carmen López", fecha: "2025-03-20", tipo: "tarea", tema: "Comprensión Lectora" },
  { asignatura: "Matemáticas", profesor: "Prof. Ana Torres", fecha: "2025-04-10", tipo: "prueba", tema: "Suma y Resta Básica" },
  { asignatura: "Ciencias", profesor: "Prof. Roberto Muñoz", fecha: "2025-04-15", tipo: "tarea", tema: "Los Seres Vivos" },
  { asignatura: "Lenguaje", profesor: "Prof. Carmen López", fecha: "2025-05-05", tipo: "prueba", tema: "Las Vocales y Consonantes" },
  { asignatura: "Matemáticas", profesor: "Prof. Ana Torres", fecha: "2025-05-12", tipo: "tarea", tema: "Figuras Geométricas" },
  { asignatura: "Historia", profesor: "Prof. Claudia Soto", fecha: "2025-05-20", tipo: "tarea", tema: "Mi Familia y Comunidad" },
  { asignatura: "Lenguaje", profesor: "Prof. Carmen López", fecha: "2025-06-03", tipo: "prueba", tema: "Lectura de Palabras" },
  { asignatura: "Matemáticas", profesor: "Prof. Ana Torres", fecha: "2025-06-10", tipo: "prueba", tema: "Números hasta el 50" },
  { asignatura: "Ciencias", profesor: "Prof. Roberto Muñoz", fecha: "2025-06-18", tipo: "tarea", tema: "El Cuerpo Humano" }
];

// Actividades para 2° Básico
const actividades2do = [
  { asignatura: "Matemáticas", profesor: "Prof. Pedro Silva", fecha: "2025-03-18", tipo: "prueba", tema: "Suma y Resta hasta el 100" },
  { asignatura: "Lenguaje", profesor: "Prof. Isabel Ramírez", fecha: "2025-03-22", tipo: "tarea", tema: "Lectura de Cuentos" },
  { asignatura: "Matemáticas", profesor: "Prof. Pedro Silva", fecha: "2025-04-12", tipo: "prueba", tema: "Multiplicación Básica" },
  { asignatura: "Ciencias", profesor: "Prof. Marcela Herrera", fecha: "2025-04-18", tipo: "tarea", tema: "Estados de la Materia" },
  { asignatura: "Lenguaje", profesor: "Prof. Isabel Ramírez", fecha: "2025-05-08", tipo: "prueba", tema: "Ortografía y Gramática" },
  { asignatura: "Matemáticas", profesor: "Prof. Pedro Silva", fecha: "2025-05-15", tipo: "tarea", tema: "Geometría Plana" },
  { asignatura: "Historia", profesor: "Prof. Andrés Lagos", fecha: "2025-05-22", tipo: "tarea", tema: "Pueblos Originarios" },
  { asignatura: "Lenguaje", profesor: "Prof. Isabel Ramírez", fecha: "2025-06-05", tipo: "prueba", tema: "Comprensión de Textos" },
  { asignatura: "Matemáticas", profesor: "Prof. Pedro Silva", fecha: "2025-06-12", tipo: "prueba", tema: "Números hasta el 200" },
  { asignatura: "Ciencias", profesor: "Prof. Marcela Herrera", fecha: "2025-06-20", tipo: "tarea", tema: "Ciclo del Agua" }
];

// Función para generar nota aleatoria entre 4.0 y 7.0
function generarNota() {
  return (Math.random() * 3 + 4.0).toFixed(1);
}

// Generar CSV
const rows = ['nombre,rut,curso,seccion,asignatura,profesor,fecha,tipo,nota,tema'];

students.forEach(student => {
  const actividades = student.curso === "1ro Básico" ? actividades1ro : actividades2do;
  
  // Ajustar profesor según sección
  const actividadesAjustadas = actividades.map(act => {
    if (student.curso === "1ro Básico") {
      if (act.asignatura === "Matemáticas") {
        return { ...act, profesor: student.seccion === "A" ? "Prof. Ana Torres" : "Prof. Luis Morales" };
      } else if (act.asignatura === "Lenguaje") {
        return { ...act, profesor: student.seccion === "A" ? "Prof. Carmen López" : "Prof. Rosa Vega" };
      }
    } else if (student.curso === "2do Básico") {
      if (act.asignatura === "Matemáticas") {
        return { ...act, profesor: student.seccion === "A" ? "Prof. Pedro Silva" : "Prof. Ricardo Díaz" };
      } else if (act.asignatura === "Lenguaje") {
        return { ...act, profesor: student.seccion === "A" ? "Prof. Isabel Ramírez" : "Prof. Marcela Fuentes" };
      }
    }
    return act;
  });
  
  actividadesAjustadas.forEach(actividad => {
    const nota = generarNota();
    rows.push(
      `${student.nombre},${student.rut},${student.curso},${student.seccion},${actividad.asignatura},${actividad.profesor},${actividad.fecha},${actividad.tipo},${nota},${actividad.tema}`
    );
  });
});

// Escribir archivo
fs.writeFileSync('calificaciones-2025-1ro-2do-basico.csv', rows.join('\n'), 'utf8');
console.log(`✅ Archivo generado con ${rows.length - 1} calificaciones`);
console.log(`   - ${students.length} estudiantes`);
console.log(`   - 10 actividades por estudiante`);
console.log(`   - Total: ${students.length * 10} registros de calificaciones`);

File: generate_grades.py
import csv
import random

# Datos base
estudiantes = [
    {"nombre": "Sofía González González", "rut": "10000000-8"},
    {"nombre": "Matías González Díaz", "rut": "10000001-6"},
    {"nombre": "Valentina González Contreras", "rut": "10000002-4"},
    {"nombre": "Benjamín González Sepúlveda", "rut": "10000003-2"},
    {"nombre": "Martina González López", "rut": "10000004-0"},
    {"nombre": "Lucas González Torres", "rut": "10000005-9"},
    {"nombre": "Isidora González Espinoza", "rut": "10000006-7"},
    {"nombre": "Agustín González Vega", "rut": "10000007-5"},
    {"nombre": "Emilia González Gutiérrez", "rut": "10000008-3"},
    {"nombre": "Tomás González Ramírez", "rut": "10000009-1"},
    {"nombre": "Amanda González Cortés", "rut": "10000010-5"},
    {"nombre": "Diego González Figueroa", "rut": "10000011-3"},
    {"nombre": "Catalina González Jara", "rut": "10000012-1"}
]

asignaturas = [
    {"nombre": "Matemáticas", "profesores": ["María López García", "Pedro Rodríguez Silva", "Ana González Muñoz"]},
    {"nombre": "Lenguaje y Comunicación", "profesores": ["Carmen López Valenzuela", "Juan García Torres", "Sofía Martínez Vega"]},
    {"nombre": "Historia, Geografía y Ciencias Sociales", "profesores": ["Roberto Fernández Castro", "Gabriela Sánchez Rojas", "Carlos Herrera Núñez"]},
    {"nombre": "Ciencias Naturales", "profesores": ["Valentina Torres Díaz", "Diego Morales Soto", "Catalina Reyes Guzmán"]},
    {"nombre": "Educación Física", "profesores": ["Sebastián Silva Morales", "Antonia Castro Campos", "Francisco Vargas Jiménez"]},
    {"nombre": "Artes Visuales", "profesores": ["Isidora Flores Paredes", "Josefa Ruiz Sepúlveda", "Manuel Romero Cortés"]},
    {"nombre": "Música", "profesores": ["Vicente Tapia Iglesias", "Nicolás Valenzuela Cruz", "Francisca Medina Aros"]},
    {"nombre": "Inglés", "profesores": ["Joaquín Araya Peña", "Maximiliano Espinoza Molina", "Renata Contreras Vera"]},
    {"nombre": "Tecnología", "profesores": ["Gabriel Vergara Pacheco", "Trinidad Santana Ibarra", "Samuel Jara Bustos"]},
    {"nombre": "Orientación", "profesores": ["Andrés Poblete Oyarzún", "Cristóbal Cortés Sandoval", "Constanza Riquelme Carvajal"]}
]

tipos_evaluacion = ["tarea", "prueba", "evaluacion"]

# Fechas distribuidas en el año 2025
# Primer semestre: Marzo a Junio (5 actividades)
fechas_sem1 = [
    "2025-03-10", "2025-03-25",  # Marzo
    "2025-04-08", "2025-04-22",  # Abril
    "2025-05-06", "2025-05-20",  # Mayo
    "2025-06-03", "2025-06-17"   # Junio
]

# Segundo semestre: Julio a Diciembre (5 actividades)
fechas_sem2 = [
    "2025-07-08", "2025-07-22",  # Julio
    "2025-08-05", "2025-08-19",  # Agosto
    "2025-09-02", "2025-09-16",  # Septiembre
    "2025-09-30",                # Septiembre
    "2025-10-14", "2025-10-28",  # Octubre
    "2025-11-11", "2025-11-25",  # Noviembre
    "2025-12-09", "2025-12-23"   # Diciembre
]

# Generar el archivo CSV
output_file = '/workspaces/superjf_v17/public/test-data/grades-consolidated-2025-COMPLETO.csv'

with open(output_file, 'w', newline='', encoding='utf-8') as csvfile:
    fieldnames = ['Nombre', 'RUT', 'Curso', 'Sección', 'Asignatura', 'Profesor', 'Fecha', 'Tipo', 'Nota']
    writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
    
    writer.writeheader()
    
    # Para cada estudiante
    for estudiante in estudiantes:
        # Para cada asignatura
        for asignatura in asignaturas:
            # Seleccionar fechas (10 en total: 5 de cada semestre)
            fechas_sel_sem1 = random.sample(fechas_sem1, 5)
            fechas_sel_sem2 = random.sample(fechas_sem2, 5)
            
            todas_fechas = sorted(fechas_sel_sem1 + fechas_sel_sem2)
            
            # Generar 10 actividades evaluativas
            for i, fecha in enumerate(todas_fechas):
                # Seleccionar tipo de evaluación (distribuido)
                tipo = tipos_evaluacion[i % 3]
                
                # Seleccionar profesor de la asignatura (rotar)
                profesor = asignatura["profesores"][i % len(asignatura["profesores"])]
                
                # Generar nota aleatoria entre 60 y 100
                nota = random.randint(60, 100)
                
                # Escribir fila
                writer.writerow({
                    'Nombre': estudiante['nombre'],
                    'RUT': estudiante['rut'],
                    'Curso': '1ro Básico',
                    'Sección': 'A',
                    'Asignatura': asignatura['nombre'],
                    'Profesor': profesor,
                    'Fecha': fecha,
                    'Tipo': tipo,
                    'Nota': nota
                })

print(f"Archivo CSV generado exitosamente: {output_file}")
print(f"Total de registros: {len(estudiantes) * len(asignaturas) * 10}")
print(f"Estudiantes: {len(estudiantes)}")
print(f"Asignaturas por estudiante: {len(asignaturas)}")
print(f"Actividades por asignatura: 10 (5 por semestre)")
